tests/test_basic_expressions.py

```diff
@@ -67,7 +67,7 @@ def exp_fn(x):
     def test_type_casting(self):
         def cast_fn(x):
             # Cast to int64 and back to float32
-            return np.float32(np.int64(x))
+            return np.astype(np.astype(x, np.int64), np.float32)
```

Comment:
This was the one place I had to change the source syntax to allow it to be overridden. There is no way (AFAIK) to overload the behavior of calling the dtype constructors on a value. However, `astype` can be overridden and is present in the array API standard.